pkg/controller/controller_flow_test.go

```diff
@@ -0,0 +1,195 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller_test
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	scfeatures "github.com/kubernetes-incubator/service-catalog/pkg/features"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+)
+
+// TestBasicFlowWithBasicAuth tests whether the controller uses correct credentials
```

Comment:
could you make this "uses correct credentials when the secret changes"?

Reply:
done